xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -248,6 +194,208 @@ private static ManagedChannel initLbChannel(
+      @Nullable
+      ObjectPool<XdsClient> xdsClientRef;
+      @Nullable
+      XdsClient xdsClient;
+      @Nullable
+      LocalityStore localityStore;
+      @Nullable
+      LoadReportClient lrsClient;
+
+      FixedXdsConfigBalancer(Helper helper) {
+        this.helper = helper;
+      }
+
+      @Override
+      public void handleNameResolutionError(Status error) {}
+
+      @Override
+      public boolean canHandleEmptyAddressListFromNameResolution() {
+        return true;
+      }
+
+      @Override
+      public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+        if (xdsClientRef != null) {
+          return;
+        }
+
+        LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
+        localityStore = localityStoreFactory.newLocalityStore(
+            helper, lbRegistry, loadStatsStore);
+        // TODO(zdapeng): Use XdsClient to do Lrs directly.
+        final LoadReportCallback lrsCallback =
+            new LoadReportCallback() {
+              @Override
+              public void onReportResponse(long reportIntervalNano) {
+                localityStore.updateOobMetricsReportInterval(reportIntervalNano);
+              }
+            };
+
+        // There are three usecases:
+        // 1. The EDS-only legacy usecase that requires balancerName from xds config.
+        // 2. The EDS-only with bootstrap usecase:
+        //    The name resolver resolves a ResolvedAddresses with an XdsConfig without balancerName
+        //    field. Use the bootstrap information to create a channel.
+        // 3. Non EDS-only usecase:
+        //    XDS_CLIENT_REF attribute is available from ResolvedAddresses either from
+        //    XdsNameResolver or CDS policy.
+        // We assume XdsConfig switching happens only within one usecase, and there is no switching
+        // between different usecases.
+        if (xdsConfig.balancerName != null) {
+          // This is the EDS-only legacy usecase that requires balancerName from xds config.
+
+          if (oldXdsConfig == null || !xdsConfig.balancerName.equals(oldXdsConfig.balancerName)) {
+            final ManagedChannel channel = initLbChannel(
+                helper, xdsConfig.balancerName, Collections.<ChannelCreds>emptyList());
+            final Node node = Node.newBuilder()
+                .setMetadata(Struct.newBuilder()
+                    .putFields(
+                        "endpoints_required",
+                        Value.newBuilder().setBoolValue(true).build()))
+                .build();
+            // TODO(zdapeng): Use XdsClient to do Lrs directly.
+            lrsClient = loadReportClientFactory.createLoadReportClient(
+                channel, helper, new ExponentialBackoffPolicy.Provider(), loadStatsStore);
+            xdsClientRef = new RefCountedXdsClientObjectPool(new XdsClientFactory() {
+              @Override
+              XdsClient createXdsClient() {
+                return new XdsComms2(
```

Comment:
if `XdsComm2` is eventually going to go away, should you have a TODO comment to replace this code?

Reply:
Done.